### ISSUE TITLE
Can not Copy  Products Variant  Mutlistore   Store A to Store B

### ISSUE THREAD

**Issue by nisteria** [PrestaShop/PrestaShop#22899]
#### Describe the bug

it is not possible to copy product variants from store A to store B.  It is only possible to create the same variants in both stores.

#### Expected behavior

Explain what you expected to happen instead.

#### Steps to Reproduce

Store A
1) open Articles with variants 
2) change the Store B
   now i have copied the article to store B, but without variants. it is also not possible to create the same variants that are already in store A. 

3) store a and store b access the same stock level

1.7.7.0 it worked 
after the update 1.7.7.1 does not work 

 

#### Additional information

* PrestaShop version: 1.7.7.1
* PHP version: 7.2.34

**Comment by hibatallahAouadni** [comment]
Hello @nisteria 

I'm a little bit confused :confused: could you please explain to me how did you copy a product from one store to another?
And I believe you mean by variants => combinations, is that so?

Thanks!

**Comment by nisteria** [comment]
![shop1](https://user-images.githubusercontent.com/1664247/105199055-9b81b100-5b3e-11eb-848f-fadaffc78ebc.jpg)
![shop2](https://user-images.githubusercontent.com/1664247/105199086-a50b1900-5b3e-11eb-8900-503dbc5e421e.jpg)
![shop3](https://user-images.githubusercontent.com/1664247/105199099-a76d7300-5b3e-11eb-936d-620cdd452366.jpg)

**Comment by nisteria** [comment]
```
InvalidArgumentException:
"" cannot be interpreted as a number

  at vendor/prestashop/decimal/src/Builder.php:40
  at PrestaShop\Decimal\Builder::parseNumber('')
     (vendor/prestashop/decimal/src/Number.php:73)
  at PrestaShop\Decimal\Number->__construct('')
     (src/Adapter/CombinationDataProvider.php:149)
  at PrestaShop\PrestaShop\Adapter\CombinationDataProvider->completeCombination(array(array('id_product_attribute' => '128', 'id_product' => '72', 'reference' => '', 'supplier_reference' => '', 'location' => '', 'ean13' => '', 'isbn' => '', 'upc' => '', 'mpn' => null, 'wholesale_price' => '0.000000', 'price' => '0.000000', 'ecotax' => '0.000000', 'quantity' => 8, 'weight' => '0.000000', 'unit_price_impact' => '0.000000', 'default_on' => null, 'minimal_quantity' => '1', 'low_stock_threshold' => null, 'low_stock_alert' => '0', 'available_date' => '0000-00-00', 'id_shop' => '3', 'id_attribute_group' => '2', 'is_color_group' => '1', 'group_name' => 'Farbe', 'attribute_name' => 'Schwarz', 'id_attribute' => '11', 'position' => '6')), object(Product))
     (src/Adapter/CombinationDataProvider.php:111)
  at PrestaShop\PrestaShop\Adapter\CombinationDataProvider->getFormCombinations(array('128', '129'), 1)
     (src/PrestaShopBundle/Controller/Admin/CombinationController.php:53)
  at PrestaShopBundle\Controller\Admin\CombinationController->generateCombinationFormAction(array('128', '129'))
     (vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php:151)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php:68)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, false)
     (vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php:200)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request), 1, false)
     (admin_Nisteria/index.php:82)
```

**Comment by nisteria** [comment]
```
in src/Adapter/CombinationDataProvider.php (line 149)
            $attribute_unity_price_impact = 1;        } elseif ($combination['unit_price_impact'] < 0) {            $attribute_unity_price_impact = -1;        }        $finalPrice = (new Number((string) $product->price))            ->plus(new Number((string) $combination['price']))            ->toPrecision(CommonAbstractType::PRESTASHOP_DECIMALS);        return [            'id_product_attribute' => $combination['id_product_attribute'],

```

**Comment by hibatallahAouadni** [comment]
Hello @nisteria 

Following your steps I didn't manage to copy a product from store to another, see the attached screen record below:

https://drive.google.com/file/d/1XCMVNsXvPMoRwYTbNw42vz1qGx7FFDOa/view

And AFAIK, copying a product from store to another in this way is impossible :confused: 
But let's ask the Product Team :wink: 

Ping @PrestaShop/product-team is there a way to copy a product from store to another?

Thanks!

**Comment by nisteria** [comment]
I have cahnge

scr/Adapter/CombinationDataProvider.php

Line 149
      $finalPrice = (new Number((string) $product->price))
            ->plus(new Number((string) $combination['price']))
            ->toPrecision(CommonAbstractType::PRESTASHOP_DECIMALS);
 to 
     /**   $finalPrice = (new Number((string) $product->price))
            ->plus(new Number((string) $combination['price']))
            ->toPrecision(CommonAbstractType::PRESTASHOP_DECIMALS);
*/

and 
Line 164 
  'final_price' => $combination['price'],

now its works

**Comment by nisteria** [comment]
now i can not SAVE the Product 
and the next problem is

Path | Message | Invalid value | Violation
-- | -- | -- | --
children[step3].children[minimal_quantity].data | Dieser Wert sollte nicht leer sein. | null | ConstraintViolation {#9034 ▼   -message: "Dieser Wert sollte nicht leer sein."   -messageTemplate: "This value should not be blank."   -parameters: [▶]   -plural: null   -root: Form {#256 ▶}   -propertyPath: "children[step3].children[minimal_quantity].data"   -invalidValue: null   -constraint: NotBlank {#933 …}   -code: "c1051bb4-d103-4f74-8988-acbcafc7fdc3"   -cause: null }

----------------------------------------------------------------------------------------------------

Message | Origin | Cause
-- | -- | --
Dieser Wert sollte nicht leer sein. | minimal_quantity | Caused by:ConstraintViolation {#9034 ▶}

 out_of_stock
 minimal_quantity     IS Empty
location
low_stock_threshold

**Comment by nisteria** [comment]
ConstraintViolation {#9034 ▼
  -message: "Dieser Wert sollte nicht leer sein."
  -messageTemplate: "This value should not be blank."
  -parameters: [▼
    "{{ value }}" => "null"
  ]
  -plural: null
  -root: Form {#256 ▼
    name: "form"
    type_class: "Symfony\Component\Form\Extension\Core\Type\FormType"
    data: [▼
      "id_product" => "72"
      "step1" => [▶]
      "step2" => [▶]
      "step3" => [▼
        "advanced_stock_management" => false
        "depends_on_stock" => null
        "qty_0" => 18.0
        "id_product_attributes" => [▶]
        "out_of_stock" => "2"
        "minimal_quantity" => null
        "location" => null
        "low_stock_threshold" => null
        "low_stock_alert" => false
        "available_now" => [▶]
        "available_later" => [▶]
        "available_date" => null
        "pack_stock_type" => 0
        "virtual_product" => [▶]
        "attributes" => ""
      ]
      "step4" => [▶]
      "step5" => [▶]
      "step6" => [▶]
      "combination_128" => [▶]
      "combination_129" => [▶]
    ]
  }

**Comment by nisteria** [comment]
"minimal_quantity" => null

**Comment by nisteria** [comment]
When i copy the Product to any Shops   the dont copy   PRICE, minimal_quantity,  Look Video 

https://drive.google.com/file/d/1k9blmGoO55elUhb_0KvATR4njTwO98tP/view?usp=sharing

**Comment by nisteria** [comment]
https://drive.google.com/file/d/1bRd521EZNwCSN7ViFp3xhAgh3d_wh-Un/view?usp=sharing

**Comment by hibatallahAouadni** [comment]
Hello @nisteria 

OK, thanks for your feedback :relaxed: 
I already notify the product team and they will respond you ASAP, so please stop spamming the issue :sweat_smile: 

Thanks!

**Comment by MatShir** [comment]
It should be fixed with 8.1 version :)

**Comment by florine2623** [comment]
This is fixed in 8.1.0.

https://github.com/PrestaShop/PrestaShop/assets/16019289/5df14c6f-cc31-444b-ae9e-0bd59f41c7be

I'm closing this issue.
